From b1dd25faabd0a1609b010540be3753e650479dde Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint.reczey@ericsson.com>
Date: Sat, 22 Aug 2009 22:46:41 +0000
Subject: [PATCH 1/2] From Frederic Peters (made endian-aware by me): fixed
 segfault with some GIOP packets

svn path=/trunk/; revision=29509

Conflicts:
	epan/dissectors/packet-giop.c

Change-Id: I464930dc6089a20faabc6e8099a716fc0e6751c4
---
 epan/dissectors/packet-giop.c | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/epan/dissectors/packet-giop.c b/epan/dissectors/packet-giop.c
index be9e005..7598392 100644
--- a/epan/dissectors/packet-giop.c
+++ b/epan/dissectors/packet-giop.c
@@ -1715,7 +1715,7 @@ static void giop_dump_collection(collection_data_t collection_type) {
  * But skip a subdissector if it has been disabled in GUI "edit protocols".
  */
 
-static gboolean try_heuristic_giop_dissector(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, int *offset,
+static gboolean try_heuristic_giop_dissector(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, guint32 *offset,
 		MessageHeader *header, gchar *operation  ) {
 
   int i,len;
@@ -1727,8 +1727,19 @@ static gboolean try_heuristic_giop_dissector(tvbuff_t *tvb, packet_info *pinfo,
 
   if (len == 0)
     return FALSE;
-  if (*offset > header->message_size)
-    return FALSE;
+  
+  {
+    guint32 message_size;
+    gboolean stream_is_big_endian = is_big_endian (header);
+
+    if (stream_is_big_endian)
+      message_size = pntohl (header->message_size);
+    else
+      message_size = pletohl (header->message_size);
+
+    if (*offset > header->message_size)
+      return FALSE;
+  }
 
   saved_proto = pinfo->current_proto;
   for (i=0; i<len; i++) {
-- 
1.9.1

